pkg/rook/purge_osd.go

```diff
@@ -37,14 +39,27 @@ func PurgeOsd(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNames
 	cephArgs := []string{
 		"auth", "print-key", "client.admin",
 	}
-	adminKey := exec.RunCommandInOperatorPod(ctx, clientsets, "ceph", cephArgs, operatorNamespace, clusterNamespace, true)
+	adminKey := pkgexec.RunCommandInOperatorPod(ctx, clientsets, "ceph", cephArgs, operatorNamespace, clusterNamespace, true)
 
-	cmd := "/bin/sh"
 	args := []string{
+		fmt.Sprintf("--namespace=%s", operatorNamespace),
+		"exec",
+		"deployments/rook-ceph-operator",
+		"--",
+		"/bin/bash",
 		"-c",
 		fmt.Sprintf("export ROOK_MON_ENDPOINTS=%s ROOK_CEPH_USERNAME=client.admin ROOK_CEPH_SECRET=%s ROOK_CONFIG_DIR=/var/lib/rook && rook ceph osd remove --osd-ids=%s --force-osd-removal=%s", monEndPoint, adminKey, osdId, flag),
 	}
 
 	logging.Info("Running purge osd command")
-	return exec.RunCommandInOperatorPod(ctx, clientsets, cmd, args, operatorNamespace, clusterNamespace, true)
```

Comment:
So this is working when you run the command directly here, but not with the `RunCommandInOperatorPod()` helper? What's different? I assume this was an experiment, and now you can find a change in `RunCommandInOperatorPod()` that will work?

Reply:
Hmm, I'm still unable to find the fix. I tried a few things but didn't help.  The problem is it working as expected until we are calling osd deployment to 0 which is not giving me any hints what the could be issue.

@BlaineEXE could you help us here?

also, I checked the rook code, and I see there also we're using `os/exec` pkg instead of creating `stream`. The stream method is only used in integration tests when we are running commands in the toolbox.

@travisn will this be a good idea to move to `os/exec` if it takes more time?

Reply:
> @travisn will this be a good idea to move to os/exec if it takes more time?

What do you mean it will take more time? It appears to me that `StreamWithContext()` is not working as expected since it doesn't stream the output from stdout. Was there a need for `StreamWithContext()`? Since the `os/exec` package is working for us, seems like we can implement it instead of the streaming method.

Are there any other commands that take a while to run in the operator, similar to the osd purge? Perhaps we didn't notice that the streaming wasn't working with other commands since they are all quick in their call to the operator.

Reply:
@travisn Okay, finally I could find the root issue, and the issue you opened in rook https://github.com/rook/rook/issues/12320 gave me a clue.

Let's discuss this in a call, meanwhile, I'll try to fix the issue soon